pkg/cloud/gcp/actuators/machine/actuator.go

```diff
@@ -47,10 +68,17 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 		machine:       machine,
 	})
 	if err != nil {
-		return fmt.Errorf(scopeFailFmt, machine.Name, err)
+		fmtErr := fmt.Errorf(scopeFailFmt, machine.Name, err)
+		return a.handleMachineError(machine, apierrors.InvalidMachineConfiguration("error decoding MachineProviderConfig: %v", fmtErr), createEventAction)
```

Comment:
`s/error decoding MachineProviderConfig/error creating machine scope` would be more appropriate

Reply:
`InvalidMachineConfiguration` and `error decoding MachineProviderConfig` are not appropriate here